pkg/util/yaml/encoder.go

```diff
@@ -0,0 +1,283 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package yaml contains a utility for marshaling an object to yaml, including field comments.
+package yaml
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+	"sort"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/gengo/parser"
+	gengotypes "k8s.io/gengo/types"
+)
+
+var durationType = reflect.TypeOf(metav1.Duration{})
+var timeType = reflect.TypeOf(metav1.Time{})
+
+type stackItem struct {
+	value    reflect.Value
+	indent   int
+	name     string
+	comments string
+	inSlice  bool
+}
+
+// Marshal writes obj to w as yaml, including field comments.
```

Comment:
is this a fork from go?

Reply:
No, I wrote it by hand

Reply:
we have the thirdparty fork package for those.

Reply:
i guess i could have done that... should i throw this out and fork go's?